google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb

```diff
@@ -66,20 +69,28 @@ def stop
           synchronize do
             return if @request_queue.nil?
             @request_queue.push self
+            @inventory.clear
```

Comment:
Won't this delete messages that need to be processed? (Or is the idea that this occurs just locally, and they will be retrieved again later?)